docs/cluster-adoption.md

```diff
@@ -0,0 +1,49 @@
+# Cluster Adoption
+
+## Adopting ClusterDeployments
+
+It is possible to adopt cluster deployments into Hive. To do so you will need to create a ClusterDeployment with Spec.Installed set to True, no Spec.Provisioning section, and include the following:
+
+  * cluster INFRAID (obtained from `oc get infrastructure cluster -o json | jq .status.infrastructureName`)
+  * cluster ID (obtained from `oc get clusterversion version -o json | jq .spec.clusterID`)
+  * reference to a properly formatted admin kubeconfig Secret:  `oc create secret generic mycluster-admin-kubeconfig --from-file=kubeconfig=/tmp/admin.kubeconfig`
+  * Spec.Platform.YourCloudProvider for your cluster, most importantly region and a properly formatted credentials Secret
+
+Use Spec.PreserveOnDelete if you do not want Hive to deprovision resources when the ClusterDeployment is deleted.
+
+### Example ClusterDeployment
+
+```yaml
+apiVersion: hive.openshift.io/v1
+kind: ClusterDeployment
+metadata:
+  name: my-gcp-cluster
+  namespace: hive
+spec:
+  baseDomain: gcp.example.com
+  clusterMetadata:
+    adminKubeconfigSecretRef:
+      name: my-gcp-cluster-admin-kubeconfig
+    clusterID: 61010205-c91d-44c9-8394-3e1790bd76f3
+    infraID: my-gcp-cluster-wsvdn
+  clusterName: my-gcp-cluster
+  installed: true
+  platform:
+    gcp:
+      credentialsSecretRef:
+        name: my-gcp-creds
+      region: us-east1
+  pullSecretRef:
+    name: pull-secret
+```
+
+## Adopting with hiveutil
+
+hiveutil is a development focused CLI tool which can be build from the hive repo. To adopt a cluster specify the following flags:
+
+```bash
+bin/hiveutil create-cluster --base-domain=example.com mycluster --adopt --adopt-admin-kubeconfig=/path/to/cluster/admin/kubeconfig --adopt-infra-id=[INFRAID] --adopt-cluster-id=[CLUSTERID]
```

Comment:
Is there an option to set `PreserveOnDelete` via hiveutil?

...looks like not :(  Should there be?

Reply:
Probably not, we agreed at the outset it was never meant for 1-1 compat with the API, and you could use -o yaml to make edits you need. I still consider this pretty fringe and not really necessary for dev focused hiveutil.